pkg/block/azure/adapter.go

```diff
@@ -196,6 +214,46 @@ func (a *Adapter) Get(ctx context.Context, obj block.ObjectPointer, _ int64) (io
 	return a.Download(ctx, obj, 0, azblob.CountToEnd)
 }
 
+func (a *Adapter) GetPreSignedURL(ctx context.Context, obj block.ObjectPointer, mode block.PreSignMode) (string, error) {
+	permissions := azblob.BlobSASPermissions{Read: true}
+	if mode == block.PreSignModeWrite {
+		permissions = azblob.BlobSASPermissions{Write: true}
+	}
+	return a.getPreSignedURL(ctx, obj, permissions)
+}
+
+func (a *Adapter) getPreSignedURL(ctx context.Context, obj block.ObjectPointer, permissions azblob.BlobSASPermissions) (string, error) {
+	qualifiedKey, err := resolveBlobURLInfo(obj)
+	if err != nil {
+		return "", err
+	}
+	vals := azblob.BlobSASSignatureValues{
+		Protocol:      azblob.SASProtocolHTTPS, // Users MUST use HTTPS (not HTTP)
+		ExpiryTime:    a.preSignedURLDurationGenerator(),
+		ContainerName: qualifiedKey.ContainerName,
+		BlobName:      qualifiedKey.BlobURL,
+		Permissions:   permissions.String(),
+	}
+
+	keyCredentials, ok := a.credentials.(*azblob.SharedKeyCredential)
+	if !ok {
+		err = fmt.Errorf("pre-signed mode on Azure is only supported for shared key credentials: %w", ErrNotImplemented)
+		a.log(ctx).WithError(err).Error("no support for pre-signed using this Azure credentials provider")
+		return "", err
+	}
+
+	sasQueryParams, err := vals.NewSASQueryParameters(keyCredentials)
+	if err != nil {
+		a.log(ctx).WithError(err).Error("could not generate SAS query parameters")
+		return "", err
+	}
+	// Create the URL of the resource you wish to access and append the SAS query parameters.
+	// Since this is a blob SAS, the URL is to the Azure storage blob.
+	qp := sasQueryParams.Encode()
+	return fmt.Sprintf(preSignedBlobPattern,
+		a.credentials, qualifiedKey.ContainerName, qualifiedKey.BlobURL, qp), nil
```

Comment:
We depends on `credentials` - string format of a type we don't control? Just making sure we do not need to call any explicit function to format the string.

Reply:
Good catch, it's supposed to be the account name extracted from the credentials! Fixed.

Reply:
@itaiad200 you caught this too, changed :)